packages/flutter_tools/lib/src/base/logger.dart

```diff
@@ -337,15 +345,31 @@ class VerboseLogger extends Logger {
   @override
   Status startProgress(
     String message, {
+    @required Duration timeout,
     String progressId,
-    bool expectSlowOperation,
-    bool multilineOutput,
-    int progressIndicatorPadding,
+    bool multilineOutput = false,
+    int progressIndicatorPadding = kDefaultStatusPadding,
   }) {
+    assert(timeout != null);
+    assert(progressIndicatorPadding != null);
     printStatus(message);
-    return Status(onFinish: () {
-      printTrace('$message (completed)');
-    })..start();
+    final Stopwatch timer = Stopwatch()..start();
+    return SilentStatus(
+      timeout: timeout,
+      onFinish: () {
+        String time;
+        if (timeout > fastOperation) {
+          time = getElapsedAsSeconds(timer.elapsed);
+        } else {
+          time = getElapsedAsMilliseconds(timer.elapsed);
+        }
+        if (timer.elapsed > timeout) {
+          printTrace('$message (completed in $time, longer than expected)');
```

Comment:
What if it doesn't complete?  Can we log two messages -- one after `timeout` has elapsed to indicate that it's taking a long time, and another when it's done to record how long it took.

Reply:
We print one when it starts and one when it ends; if the end one didn't print then it should be clear that it didn't complete. Not sure what value the intermediate one would be in a verbose log: verbose logs are generally examined post-hoc, not live.